base/tests/test_MAPL_ISO8601_Time.pf

```diff
@@ -0,0 +1,486 @@
+#include "MAPL_Exceptions.h"
+
+module test_MAPL_ISO8601_Time
+      use MAPL_ExceptionHandling
+      use MAPL_ISO8601_Time
+      use pfunit
+      implicit none
+
+      integer, dimension(12), parameter :: ENDS = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+      integer, dimension(size(ENDS)), parameter :: ENDS_LEAP = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+
+contains
+
+   @test
+   subroutine test_date_fields()
+      integer :: y
+      integer :: m
+      integer :: d
+      type(date_fields) :: df
+     
+      y = 2022
+      m = 06
+      d = 27
+     
+      df = date_fields(y, m, d)
+
+      @assertEqual(y, df % year)
+      @assertEqual(m, df % month)
+      @assertEqual(d, df % day)
+
+   end subroutine test_date_fields
+
+   @test
+   subroutine test_date_fields_custom_constructor()
+      type(date_fields) :: df
+      integer, dimension(3) :: f = [2022, 06, 27]
+     
+      df = date_fields(f)
+
+      @assertEqual(f(1), df % year)
+      @assertEqual(f(2), df % month)
+      @assertEqual(f(3), df % day)
+
+   end subroutine test_date_fields_custom_constructor
+
+   @test
+   subroutine test_time_fields()
+      integer :: h
+      integer :: m
+      integer :: s
+      integer :: ms
+      type(time_fields) :: tf
+     
+      h = 13
+      m = 02
+      s = 27
+      ms = 37
+     
+      tf = time_fields(h, m, s, ms)
+
+      @assertEqual(h, tf % hour)
+      @assertEqual(m, tf % minute)
+      @assertEqual(s, tf % second)
+      @assertEqual(ms, tf % millisecond)
```

Comment:
Is this just testing that the constructor copies the data?   If so, probably not worth the effort.